webui/react/src/components/ActionDropdown.tsx

```diff
@@ -0,0 +1,93 @@
+import { Dropdown, Menu, Modal, ModalFuncProps } from 'antd';
+import { MenuInfo } from 'rc-menu/lib/interface';
+import React, { JSXElementConstructor } from 'react';
+
+import Icon from 'components/Icon';
+import handleError, { ErrorLevel, ErrorType } from 'ErrorHandler';
+import { capitalize } from 'utils/string';
+import { Eventually } from 'utils/types';
+
+import css from './ActionDropdown.module.scss';
+
+// TODO parameterize Action using Enums? https://github.com/microsoft/TypeScript/issues/30611
```

Comment:
**suggestion:** I was going to suggest using an enum instead as well. I redefined your `JobAction` to be:
```
export enum JobAction {
  Cancel = 'Cancel',
  Kill = 'Kill',
  ManageJob = 'Manage Job',
  MoveToTop = 'Move to Top',
}
```

And changed the JSX of `JobQueue` to be:
```
              <ActionDropdown<JobAction>
                actionOrder={[
                  JobAction.MoveToTop,
                  JobAction.ManageJob,
                  JobAction.Cancel,
                  JobAction.Kill,
                ]}
                confirmations={{
                  [JobAction.Cancel]: { cancelText: 'Abort' },
                  [JobAction.Kill]: {},
                  [JobAction.MoveToTop]: {},
                }}
                thing={{ id: record.name, kind: 'job' }}
                onTrigger={dropDownOnTrigger(record)}
              />
```

and it worked great

Reply:
I pulled actiondropdown into its own pr https://github.com/determined-ai/determined/pull/3194 and addressed it here